### PR TITLE
qol: Replace string with fixed size buffer

### PR DESCRIPTION
**Pull request by maidesu**
All my homies use fixed length arrays

### CODE REVIEW

**Comment by minisbett** [comment]
Very interesting approach, haven't seen the function readBytesUntil. Although could you elaborate on why you added a definition for the size of the input buffers?

**Comment by maidesu** [comment]
Tl;dr from discord convo:

`readBytesUntil()` `char command[]` `char arg0[]` `char keyStr[]` `char setting[]`
all depend on the length of `char buff[]` as a "worst case" size - their contents will always be
less than or equal to the buffer itself.

**Comment by minisbett** [comment]
Is this still a draft?

**Comment by maidesu** [comment]
No more changes planned here until testing

File: include/definitions.hpp [hunk 1]
```diff
@@ -26,6 +26,9 @@
 // This value is important to reset the rapid trigger state properly with continuous rapid trigger.
 #define CONTINUOUS_RAPID_TRIGGER_THRESHOLD 10
 
+// Size of the array which the serial buffer is copied to
+#define INPUT_BUFF_SIZE 64
```
Comment (minisbett) [inline include/definitions.hpp]: Could you change this to 256?

File: src/handlers/serial_handler.cpp [hunk 2]
```diff
@@ -11,20 +11,18 @@ extern "C"
 // Define a handy macro for printing with a newline character at the end.
 #define print(fmt, ...) Serial.printf(fmt "\n", __VA_ARGS__)
 
-void SerialHandler::handleSerialInput(String *inputStr)
+void SerialHandler::handleSerialInput(char *buff)
```
Comment (minisbett) [inline src/handlers/serial_handler.cpp]: Could you rename this to something like "input"? I think that name is a better indicator for it's purpose than "buff"

File: src/main.cpp [hunk 3]
```diff
@@ -26,10 +26,14 @@ void setup()
 void loop()
 {
     // Check for any serial commands received.
-    while (Serial.available())
+    while (Serial.available() > 0)
```
Comment (minisbett) [inline src/main.cpp:29]: What's this > 0 check for?

**Review by minisbett — CHANGES_REQUESTED** [review]
Requested some changes and left some questions.

File: src/main.cpp (hunk 3 above)
Comment (maidesu) [reply src/main.cpp:29]: > Returns
> The number of bytes available to read.

The return type int from the function `Serial.available()` could technically make the expression
`( Serial.available() )` true for negative integers while `( Serial.available() > 0 )` is obviously only true for
positive integers.

See [Serial.available()](https://reference.arduino.cc/reference/en/language/functions/communication/serial/available/)

Comment (minisbett) [reply src/main.cpp:29]: hmm I get the idea

**Comment by minisbett** [comment]
Could you solve merge conflicts and check whether recently added code needs to be adjusted as well?

**Review by minisbett — CHANGES_REQUESTED** [review]
https://github.com/minipadKB/minipad-firmware/pull/17#issuecomment-1541511790

**Comment by minisbett** [comment]
Replaced by #42

**Pull request closed without merging**